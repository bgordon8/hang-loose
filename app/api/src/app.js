import express from 'express'

const app = express()

app.get("/", (req, res) => {
    res.status(200).json({
        message: "hello world!!!! thats whats up!!!"
    })
})

if (process.env.NODE_ENV !== "test") {
    app.listen(3000, () => {
        console.log('listening on port 3000...')
    })
}

export default app