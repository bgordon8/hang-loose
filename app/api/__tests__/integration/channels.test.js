import supertest from 'supertest'
import app from '../../src/app'

const req = supertest(app)

describe('routes : channels', () => {
  describe('GET /channels', () => {
    it('returns all channels', async () => {
      const res = await req.get('/channels')

      expect(res.status).toBe(200)
      expect(res.body).toHaveProperty('status')
      expect(res.body.status).toBe('success')
      expect(res.body).toHaveProperty('channels')
      expect(res.body.channels[0]).toHaveProperty('id')
      expect(res.body.channels[0].id).toBe(1)
      expect(res.body.channels[0]).toHaveProperty('workspaceId')
      expect(res.body.channels[0].workspaceId).toBe(1)
      expect(res.body.channels[0]).toHaveProperty('name')
      expect(res.body.channels[0].name).toBe('general')
      expect(res.body).toMatchSnapshot()
    })
  })

  describe('GET /channels/:id', () => {
    it('returns channel by id', async () => {
      const res = await req.get('/channels/1')

      expect(res.status).toBe(200)
      expect(res.body).toHaveProperty('status')
      expect(res.body.status).toBe('success')
      expect(res.body).toHaveProperty('channel')
      expect(res.body.channel).toHaveProperty('id')
      expect(res.body.channel.id).toBe(1)
      expect(res.body.channel).toHaveProperty('workspaceId')
      expect(res.body.channel.workspaceId).toBe(1)
      expect(res.body.channel).toHaveProperty('name')
      expect(res.body.channel.name).toBe('general')
      expect(res.body).toMatchSnapshot()
    })
  })

  describe('POST /channels', () => {
    it('creates new channel', async () => {
      const res = await req.post('/channels').send({
        name: 'testchannel',
        workspaceId: 1,
      })

      expect(res.status).toBe(200)
      expect(res.body).toHaveProperty('status')
      expect(res.body.status).toBe('success')
      expect(res.body).toHaveProperty('channel')
      expect(res.body.channel).toHaveProperty('id')
      expect(res.body.channel.id).toBe(4)
      expect(res.body.channel).toHaveProperty('name')
      expect(res.body.channel.name).toBe('testchannel')
      expect(res.body).toMatchSnapshot({
        channel: {
          created_at: expect.any(String),
          updated_at: expect.any(String),
        },
      })
    })
  })

  describe('PUT /channels/:id', () => {
    it('updates channel by id', async () => {
      const res = await req.put('/channels/1').send({
        name: 'updatedname',
      })

      expect(res.status).toBe(200)
      expect(res.body).toHaveProperty('status')
      expect(res.body.status).toBe('success')
      expect(res.body).toHaveProperty('channel')
      expect(res.body.channel).toHaveProperty('id')
      expect(res.body.channel.id).toBe(1)
      expect(res.body.channel).toHaveProperty('name')
      expect(res.body.channel.name).toBe('updatedname')
      expect(res.body).toMatchSnapshot()
    })
  })

  describe('DELETE /channels/:id', () => {
    it('deletes channel by id', async () => {
      const res = await req.del('/channels/1')

      expect(res.status).toBe(200)
      expect(res.body).toHaveProperty('status')
      expect(res.body.status).toBe('success')
      expect(res.body).toHaveProperty('channel')
      expect(res.body.channel).toHaveProperty('id')
      expect(res.body.channel.id).toBe(1)
      expect(res.body.channel).toHaveProperty('name')
      expect(res.body.channel.name).toBe('general')
      expect(res.body.channel).toHaveProperty('workspaceId')
      expect(res.body.channel.workspaceId).toBe(1)
      expect(res.body).toMatchSnapshot()
    })
  })
})
