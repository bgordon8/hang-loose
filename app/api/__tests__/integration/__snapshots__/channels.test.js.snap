// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes : channels DELETE /channels/:id deletes channel by id 1`] = `
Object {
  "channel": Object {
    "created_at": "2020-10-21T06:00:00.000Z",
    "id": 1,
    "name": "general",
    "public": true,
    "updated_at": "2020-10-21T06:00:00.000Z",
    "workspaceId": 1,
  },
  "status": "success",
}
`;

exports[`routes : channels GET /channels returns all channels 1`] = `
Object {
  "channels": Array [
    Object {
      "created_at": "2020-10-21T06:00:00.000Z",
      "id": 1,
      "name": "general",
      "public": true,
      "updated_at": "2020-10-21T06:00:00.000Z",
      "workspaceId": 1,
    },
    Object {
      "created_at": "2020-10-21T06:00:00.000Z",
      "id": 2,
      "name": "random",
      "public": true,
      "updated_at": "2020-10-21T06:00:00.000Z",
      "workspaceId": 1,
    },
    Object {
      "created_at": "2020-10-21T06:00:00.000Z",
      "id": 3,
      "name": "planned",
      "public": true,
      "updated_at": "2020-10-21T06:00:00.000Z",
      "workspaceId": 2,
    },
  ],
  "status": "success",
}
`;

exports[`routes : channels GET /channels/:id returns channel by id 1`] = `
Object {
  "channel": Object {
    "created_at": "2020-10-21T06:00:00.000Z",
    "id": 1,
    "name": "general",
    "public": true,
    "updated_at": "2020-10-21T06:00:00.000Z",
    "workspaceId": 1,
  },
  "status": "success",
}
`;

exports[`routes : channels POST /channels creates new channel 1`] = `
Object {
  "channel": Object {
    "created_at": Any<String>,
    "id": 4,
    "name": "testchannel",
    "public": true,
    "updated_at": Any<String>,
    "workspaceId": 1,
  },
  "status": "success",
}
`;

exports[`routes : channels PUT /channels/:id updates channel by id 1`] = `
Object {
  "channel": Object {
    "created_at": "2020-10-21T06:00:00.000Z",
    "id": 1,
    "name": "updatedname",
    "public": true,
    "updated_at": "2020-10-21T06:00:00.000Z",
    "workspaceId": 1,
  },
  "status": "success",
}
`;
